### PR TITLE
Use only PublicIpAddress in Azure Firewall, not both Public and InternalPublic

### PR DESCRIPTION
**Pull request by fanymanea**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by lmazuel** [comment]
Hi @fanymanea 
For what I understand, this is a Swagger fix that should be cross-api-version. Could you please do another PR to "master" fixing all api-versions where this fix applies?
Thanks!

File: specification/network/resource-manager/Microsoft.Network/stable/2018-08-01/azureFirewall.json [hunk 1]
```diff
@@ -259,22 +259,18 @@
...
+              "description": "The Firewall Internal Load Balancer IP to be used as the next hop in User Defined Routes."
+            },
+            "subnet":{
                "$ref":"./network.json#/definitions/SubResource",
                "description":"Reference of the subnet resource. This resource must be named 'AzureFirewallSubnet'."
             },
-			"internalPublicIpAddress":{  
-               "$ref":"./network.json#/definitions/SubResource",
-               "description":"Reference of the PublicIP resource. This field is a mandatory input."
+            "publicIPAddress": {
```
Comment (lmazuel) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-08-01/azureFirewall.json]: You wrote here `publicIPAddress`, but `publicIpAddress`. JSON is case-sensitive, please be sure to put EXACTLY what the endpoint is returning.

**Review by lmazuel — CHANGES_REQUESTED** [review]
Incompatible case between spec and examples

**Comment by fanymanea** [comment]
@lmazuel, unfortunately we were not allowed to change this for previous API versions in NRP. It would have been a lot easier for us to change it for all versions.
We were not even allowed to support both Internal and Public IPs for newer API versions.

**Comment by lmazuel** [comment]
@fanymanea , so you're telling that 2018-08-01 introduces a breaking change, and that `internalPublicIpAddress` will not exist anymore? What's the story for customers moving from 2018-07-01 to 2018-08-01 then?

**Comment by fanymanea** [comment]
@lmazuel The answer is yes for the breaking change question.
And for customers moving to 08-01 version, they need to stop renaming the field. But for older APIs, they must continue to rename the field between GET and PUT.

**Review by lmazuel — APPROVED** [review]

**Pull request merged**